Feature 33 GitHub

JohnHalleyGotway opened the pull request:
This PR includes the following 3 changes:
- change github.com/NCAR to github.com/dtcenter
- change ncar.github.io to dtcenter.github.io
- all references of METdb to METdatadb

Please review the changes carefully before accepting. I wonder if changing METdb to METdatadb will cause any problems in the code.

venitahagerty commented:
I have reviewed the files in dtcenter/METdatadb. I updated some README
files and some comments in code. It looks good to me.

On Wed, Aug 19, 2020 at 4:45 PM johnhg <notifications@github.com> wrote:

> @JohnHalleyGotway <https://github.com/JohnHalleyGotway> requested your
> review on: #34 <https://github.com/dtcenter/METdatadb/pull/34> Feature 33
> GitHub.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METdatadb/pull/34#event-3672860800>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJGI4XVWZZAUGB5R6MQ4F6TSBRIZFANCNFSM4QFNWKQQ>
> .
>


-- 
Venita Hagerty, PhD
Research Associate
CIRA at NOAA, GSL/AVID
2B152, x6098

JohnHalleyGotway commented:
Venita,

Great. If you're happy with the changes, can you please approve the Pull
Request? And then you could click the "Squash and Merge" button which will
merge these changes into your master branch.

This is the process we've been following for development in the other
METplus repos.
(1) Write up an issue to describe the new work.
(2) Create a feature branch for that work that includes the issue number
(e.g. feature_33_github).
(3) Do the work in the feature branch.
(4) Submit a pull request to merge those changes back into the source
branch.
(5) Once the PR is approved, merge the changes and delete the feature
branch.

So we're at steps (4) and (5) here. If you'd prefer to meet through Google
and share screens for this, please just let me know.

Thanks,
John

On Thu, Aug 20, 2020 at 11:29 AM Venita Hagerty <notifications@github.com>
wrote:

> I have reviewed the files in dtcenter/METdatadb. I updated some README
> files and some comments in code. It looks good to me.
>
> On Wed, Aug 19, 2020 at 4:45 PM johnhg <notifications@github.com> wrote:
>
> > @JohnHalleyGotway <https://github.com/JohnHalleyGotway> requested your
> > review on: #34 <https://github.com/dtcenter/METdatadb/pull/34> Feature
> 33
> > GitHub.
> >
> > —
> > You are receiving this because your review was requested.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/dtcenter/METdatadb/pull/34#event-3672860800>, or
> > unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AJGI4XVWZZAUGB5R6MQ4F6TSBRIZFANCNFSM4QFNWKQQ
> >
> > .
> >
>
>
> --
> Venita Hagerty, PhD
> Research Associate
> CIRA at NOAA, GSL/AVID
> 2B152, x6098
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677798966>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFA4HKD4R76WPSIR47VUZLTSBVMQHANCNFSM4QFNWKQQ>
> .
>

venitahagerty commented:
Hi John,

Since I was the only person on the repository for a while, and since I
haven't worked with Pull requests before, I haven't been using them. I may
need help with that.

I have been making feature branches with the issue number, but I didn't
realize you'd made one for renaming the repository.

Also, I didn't think we were renaming METdbLoad.py. I thought the two teams
discussed that, and Tara was going to tell us if we needed to, but she
wasn't expecting that we would.

Thank you,
Venita

On Thu, Aug 20, 2020 at 12:06 PM johnhg <notifications@github.com> wrote:

> Venita,
>
> Great. If you're happy with the changes, can you please approve the Pull
> Request? And then you could click the "Squash and Merge" button which will
> merge these changes into your master branch.
>
> This is the process we've been following for development in the other
> METplus repos.
> (1) Write up an issue to describe the new work.
> (2) Create a feature branch for that work that includes the issue number
> (e.g. feature_33_github).
> (3) Do the work in the feature branch.
> (4) Submit a pull request to merge those changes back into the source
> branch.
> (5) Once the PR is approved, merge the changes and delete the feature
> branch.
>
> So we're at steps (4) and (5) here. If you'd prefer to meet through Google
> and share screens for this, please just let me know.
>
> Thanks,
> John
>
> On Thu, Aug 20, 2020 at 11:29 AM Venita Hagerty <notifications@github.com>
> wrote:
>
> > I have reviewed the files in dtcenter/METdatadb. I updated some README
> > files and some comments in code. It looks good to me.
> >
> > On Wed, Aug 19, 2020 at 4:45 PM johnhg <notifications@github.com> wrote:
> >
> > > @JohnHalleyGotway <https://github.com/JohnHalleyGotway> requested your
> > > review on: #34 <https://github.com/dtcenter/METdatadb/pull/34> Feature
> > 33
> > > GitHub.
> > >
> > > —
> > > You are receiving this because your review was requested.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/dtcenter/METdatadb/pull/34#event-3672860800>, or
> > > unsubscribe
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/AJGI4XVWZZAUGB5R6MQ4F6TSBRIZFANCNFSM4QFNWKQQ
> > >
> > > .
> > >
> >
> >
> > --
> > Venita Hagerty, PhD
> > Research Associate
> > CIRA at NOAA, GSL/AVID
> > 2B152, x6098
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677798966>,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AFA4HKD4R76WPSIR47VUZLTSBVMQHANCNFSM4QFNWKQQ
> >
> > .
> >
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677816749>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJGI4XW62GJQLK7XVPF6UALSBVQ3DANCNFSM4QFNWKQQ>
> .
>


-- 
Venita Hagerty, PhD
Research Associate
CIRA at NOAA, GSL/AVID
2B152, x6098

venitahagerty commented:
I also hadn't updated to Master for a while, and I'm wondering if you used
master as your base?

Venita

On Thu, Aug 20, 2020 at 1:22 PM Venita Hagerty - NOAA Affiliate <
venita.hagerty@noaa.gov> wrote:

> Hi John,
>
> Since I was the only person on the repository for a while, and since I
> haven't worked with Pull requests before, I haven't been using them. I may
> need help with that.
>
> I have been making feature branches with the issue number, but I didn't
> realize you'd made one for renaming the repository.
>
> Also, I didn't think we were renaming METdbLoad.py. I thought the two
> teams discussed that, and Tara was going to tell us if we needed to, but
> she wasn't expecting that we would.
>
> Thank you,
> Venita
>
> On Thu, Aug 20, 2020 at 12:06 PM johnhg <notifications@github.com> wrote:
>
>> Venita,
>>
>> Great. If you're happy with the changes, can you please approve the Pull
>> Request? And then you could click the "Squash and Merge" button which will
>> merge these changes into your master branch.
>>
>> This is the process we've been following for development in the other
>> METplus repos.
>> (1) Write up an issue to describe the new work.
>> (2) Create a feature branch for that work that includes the issue number
>> (e.g. feature_33_github).
>> (3) Do the work in the feature branch.
>> (4) Submit a pull request to merge those changes back into the source
>> branch.
>> (5) Once the PR is approved, merge the changes and delete the feature
>> branch.
>>
>> So we're at steps (4) and (5) here. If you'd prefer to meet through Google
>> and share screens for this, please just let me know.
>>
>> Thanks,
>> John
>>
>> On Thu, Aug 20, 2020 at 11:29 AM Venita Hagerty <notifications@github.com
>> >
>> wrote:
>>
>> > I have reviewed the files in dtcenter/METdatadb. I updated some README
>> > files and some comments in code. It looks good to me.
>> >
>> > On Wed, Aug 19, 2020 at 4:45 PM johnhg <notifications@github.com>
>> wrote:
>> >
>> > > @JohnHalleyGotway <https://github.com/JohnHalleyGotway> requested
>> your
>> > > review on: #34 <https://github.com/dtcenter/METdatadb/pull/34>
>> Feature
>> > 33
>> > > GitHub.
>> > >
>> > > —
>> > > You are receiving this because your review was requested.
>> > > Reply to this email directly, view it on GitHub
>> > > <https://github.com/dtcenter/METdatadb/pull/34#event-3672860800>, or
>> > > unsubscribe
>> > > <
>> >
>> https://github.com/notifications/unsubscribe-auth/AJGI4XVWZZAUGB5R6MQ4F6TSBRIZFANCNFSM4QFNWKQQ
>> > >
>> > > .
>> > >
>> >
>> >
>> > --
>> > Venita Hagerty, PhD
>> > Research Associate
>> > CIRA at NOAA, GSL/AVID
>> > 2B152, x6098
>> >
>> > —
>> > You are receiving this because you were mentioned.
>> > Reply to this email directly, view it on GitHub
>> > <https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677798966>,
>> > or unsubscribe
>> > <
>> https://github.com/notifications/unsubscribe-auth/AFA4HKD4R76WPSIR47VUZLTSBVMQHANCNFSM4QFNWKQQ
>> >
>> > .
>> >
>>
>> —
>> You are receiving this because your review was requested.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677816749>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AJGI4XW62GJQLK7XVPF6UALSBVQ3DANCNFSM4QFNWKQQ>
>> .
>>
>
>
> --
> Venita Hagerty, PhD
> Research Associate
> CIRA at NOAA, GSL/AVID
> 2B152, x6098
>


-- 
Venita Hagerty, PhD
Research Associate
CIRA at NOAA, GSL/AVID
2B152, x6098

JohnHalleyGotway commented:
Venita,

I did see that my sed commands had changed some names like that. And I
wasn't totally sure about that.

And I totally understand not using PR's when you're working alone. No need
really. I just wanted to submit my changes in this way so that you'd review
them.

I'm glad to see you have a "develop" branch in there. Let me redo this PR
with a branch from develop instead.

And to avoid confusion, can we please just remove the master branch? We
don't have a master in the other METplus repos (or at least we shouldn't).
Instead, we have develop for new work going into the next release, and
master_vX.Y for each stable release. So when you do a v1.0 release, you'd
copy develop over into master_v1.0 and then you'd commit any bufixes for
v1.0 into both master_v1.0 (for a potential v1.0.1 bugfix release) and the
develop branch (for the next version).

Does that all make sense?

Thanks,
John

On Thu, Aug 20, 2020 at 1:31 PM Venita Hagerty <notifications@github.com>
wrote:

> I also hadn't updated to Master for a while, and I'm wondering if you used
> master as your base?
>
> Venita
>
> On Thu, Aug 20, 2020 at 1:22 PM Venita Hagerty - NOAA Affiliate <
> venita.hagerty@noaa.gov> wrote:
>
> > Hi John,
> >
> > Since I was the only person on the repository for a while, and since I
> > haven't worked with Pull requests before, I haven't been using them. I
> may
> > need help with that.
> >
> > I have been making feature branches with the issue number, but I didn't
> > realize you'd made one for renaming the repository.
> >
> > Also, I didn't think we were renaming METdbLoad.py. I thought the two
> > teams discussed that, and Tara was going to tell us if we needed to, but
> > she wasn't expecting that we would.
> >
> > Thank you,
> > Venita
> >
> > On Thu, Aug 20, 2020 at 12:06 PM johnhg <notifications@github.com>
> wrote:
> >
> >> Venita,
> >>
> >> Great. If you're happy with the changes, can you please approve the Pull
> >> Request? And then you could click the "Squash and Merge" button which
> will
> >> merge these changes into your master branch.
> >>
> >> This is the process we've been following for development in the other
> >> METplus repos.
> >> (1) Write up an issue to describe the new work.
> >> (2) Create a feature branch for that work that includes the issue number
> >> (e.g. feature_33_github).
> >> (3) Do the work in the feature branch.
> >> (4) Submit a pull request to merge those changes back into the source
> >> branch.
> >> (5) Once the PR is approved, merge the changes and delete the feature
> >> branch.
> >>
> >> So we're at steps (4) and (5) here. If you'd prefer to meet through
> Google
> >> and share screens for this, please just let me know.
> >>
> >> Thanks,
> >> John
> >>
> >> On Thu, Aug 20, 2020 at 11:29 AM Venita Hagerty <
> notifications@github.com
> >> >
> >> wrote:
> >>
> >> > I have reviewed the files in dtcenter/METdatadb. I updated some README
> >> > files and some comments in code. It looks good to me.
> >> >
> >> > On Wed, Aug 19, 2020 at 4:45 PM johnhg <notifications@github.com>
> >> wrote:
> >> >
> >> > > @JohnHalleyGotway <https://github.com/JohnHalleyGotway> requested
> >> your
> >> > > review on: #34 <https://github.com/dtcenter/METdatadb/pull/34>
> >> Feature
> >> > 33
> >> > > GitHub.
> >> > >
> >> > > —
> >> > > You are receiving this because your review was requested.
> >> > > Reply to this email directly, view it on GitHub
> >> > > <https://github.com/dtcenter/METdatadb/pull/34#event-3672860800>,
> or
> >> > > unsubscribe
> >> > > <
> >> >
> >>
> https://github.com/notifications/unsubscribe-auth/AJGI4XVWZZAUGB5R6MQ4F6TSBRIZFANCNFSM4QFNWKQQ
> >> > >
> >> > > .
> >> > >
> >> >
> >> >
> >> > --
> >> > Venita Hagerty, PhD
> >> > Research Associate
> >> > CIRA at NOAA, GSL/AVID
> >> > 2B152, x6098
> >> >
> >> > —
> >> > You are receiving this because you were mentioned.
> >> > Reply to this email directly, view it on GitHub
> >> > <https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677798966
> >,
> >> > or unsubscribe
> >> > <
> >>
> https://github.com/notifications/unsubscribe-auth/AFA4HKD4R76WPSIR47VUZLTSBVMQHANCNFSM4QFNWKQQ
> >> >
> >> > .
> >> >
> >>
> >> —
> >> You are receiving this because your review was requested.
> >> Reply to this email directly, view it on GitHub
> >> <https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677816749>,
> >> or unsubscribe
> >> <
> https://github.com/notifications/unsubscribe-auth/AJGI4XW62GJQLK7XVPF6UALSBVQ3DANCNFSM4QFNWKQQ
> >
> >> .
> >>
> >
> >
> > --
> > Venita Hagerty, PhD
> > Research Associate
> > CIRA at NOAA, GSL/AVID
> > 2B152, x6098
> >
>
>
> --
> Venita Hagerty, PhD
> Research Associate
> CIRA at NOAA, GSL/AVID
> 2B152, x6098
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677857757>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFA4HKCJQUS7UZ7KGNWFPUDSBV2YHANCNFSM4QFNWKQQ>
> .
>

JohnHalleyGotway commented:
Closing this PR without merging. Need to redo using a branch from develop, not master. And be sure not to change METdbLoad to METdatadbLoad! That should remain unchanged.

venitahagerty commented:
I've been trying to model what I do after the METviewer repository. It
looks like that's transitioning to what you're describing. Is this
described in the developer notes somewhere so I can refer back to it?

I think I understand, and it sounds clearer without a single master branch.
When I do a release, I make sure the develop branch is completely ready,
and then I make a copy of it named as master with the release number. Over
time, as I have multiple releases, I'll wind up with multiple
master+release# branches. And I could remove the current master branch, and
make develop the default.

Thank you,
Venita

On Thu, Aug 20, 2020 at 2:09 PM johnhg <notifications@github.com> wrote:

> Venita,
>
> I did see that my sed commands had changed some names like that. And I
> wasn't totally sure about that.
>
> And I totally understand not using PR's when you're working alone. No need
> really. I just wanted to submit my changes in this way so that you'd review
> them.
>
> I'm glad to see you have a "develop" branch in there. Let me redo this PR
> with a branch from develop instead.
>
> And to avoid confusion, can we please just remove the master branch? We
> don't have a master in the other METplus repos (or at least we shouldn't).
> Instead, we have develop for new work going into the next release, and
> master_vX.Y for each stable release. So when you do a v1.0 release, you'd
> copy develop over into master_v1.0 and then you'd commit any bufixes for
> v1.0 into both master_v1.0 (for a potential v1.0.1 bugfix release) and the
> develop branch (for the next version).
>
> Does that all make sense?
>
> Thanks,
> John
>
> On Thu, Aug 20, 2020 at 1:31 PM Venita Hagerty <notifications@github.com>
> wrote:
>
> > I also hadn't updated to Master for a while, and I'm wondering if you
> used
> > master as your base?
> >
> > Venita
> >
> > On Thu, Aug 20, 2020 at 1:22 PM Venita Hagerty - NOAA Affiliate <
> > venita.hagerty@noaa.gov> wrote:
> >
> > > Hi John,
> > >
> > > Since I was the only person on the repository for a while, and since I
> > > haven't worked with Pull requests before, I haven't been using them. I
> > may
> > > need help with that.
> > >
> > > I have been making feature branches with the issue number, but I didn't
> > > realize you'd made one for renaming the repository.
> > >
> > > Also, I didn't think we were renaming METdbLoad.py. I thought the two
> > > teams discussed that, and Tara was going to tell us if we needed to,
> but
> > > she wasn't expecting that we would.
> > >
> > > Thank you,
> > > Venita
> > >
> > > On Thu, Aug 20, 2020 at 12:06 PM johnhg <notifications@github.com>
> > wrote:
> > >
> > >> Venita,
> > >>
> > >> Great. If you're happy with the changes, can you please approve the
> Pull
> > >> Request? And then you could click the "Squash and Merge" button which
> > will
> > >> merge these changes into your master branch.
> > >>
> > >> This is the process we've been following for development in the other
> > >> METplus repos.
> > >> (1) Write up an issue to describe the new work.
> > >> (2) Create a feature branch for that work that includes the issue
> number
> > >> (e.g. feature_33_github).
> > >> (3) Do the work in the feature branch.
> > >> (4) Submit a pull request to merge those changes back into the source
> > >> branch.
> > >> (5) Once the PR is approved, merge the changes and delete the feature
> > >> branch.
> > >>
> > >> So we're at steps (4) and (5) here. If you'd prefer to meet through
> > Google
> > >> and share screens for this, please just let me know.
> > >>
> > >> Thanks,
> > >> John
> > >>
> > >> On Thu, Aug 20, 2020 at 11:29 AM Venita Hagerty <
> > notifications@github.com
> > >> >
> > >> wrote:
> > >>
> > >> > I have reviewed the files in dtcenter/METdatadb. I updated some
> README
> > >> > files and some comments in code. It looks good to me.
> > >> >
> > >> > On Wed, Aug 19, 2020 at 4:45 PM johnhg <notifications@github.com>
> > >> wrote:
> > >> >
> > >> > > @JohnHalleyGotway <https://github.com/JohnHalleyGotway> requested
> > >> your
> > >> > > review on: #34 <https://github.com/dtcenter/METdatadb/pull/34>
> > >> Feature
> > >> > 33
> > >> > > GitHub.
> > >> > >
> > >> > > —
> > >> > > You are receiving this because your review was requested.
> > >> > > Reply to this email directly, view it on GitHub
> > >> > > <https://github.com/dtcenter/METdatadb/pull/34#event-3672860800>,
> > or
> > >> > > unsubscribe
> > >> > > <
> > >> >
> > >>
> >
> https://github.com/notifications/unsubscribe-auth/AJGI4XVWZZAUGB5R6MQ4F6TSBRIZFANCNFSM4QFNWKQQ
> > >> > >
> > >> > > .
> > >> > >
> > >> >
> > >> >
> > >> > --
> > >> > Venita Hagerty, PhD
> > >> > Research Associate
> > >> > CIRA at NOAA, GSL/AVID
> > >> > 2B152, x6098
> > >> >
> > >> > —
> > >> > You are receiving this because you were mentioned.
> > >> > Reply to this email directly, view it on GitHub
> > >> > <
> https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677798966
> > >,
> > >> > or unsubscribe
> > >> > <
> > >>
> >
> https://github.com/notifications/unsubscribe-auth/AFA4HKD4R76WPSIR47VUZLTSBVMQHANCNFSM4QFNWKQQ
> > >> >
> > >> > .
> > >> >
> > >>
> > >> —
> > >> You are receiving this because your review was requested.
> > >> Reply to this email directly, view it on GitHub
> > >> <https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677816749
> >,
> > >> or unsubscribe
> > >> <
> >
> https://github.com/notifications/unsubscribe-auth/AJGI4XW62GJQLK7XVPF6UALSBVQ3DANCNFSM4QFNWKQQ
> > >
> > >> .
> > >>
> > >
> > >
> > > --
> > > Venita Hagerty, PhD
> > > Research Associate
> > > CIRA at NOAA, GSL/AVID
> > > 2B152, x6098
> > >
> >
> >
> > --
> > Venita Hagerty, PhD
> > Research Associate
> > CIRA at NOAA, GSL/AVID
> > 2B152, x6098
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677857757>,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AFA4HKCJQUS7UZ7KGNWFPUDSBV2YHANCNFSM4QFNWKQQ
> >
> > .
> >
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677877914>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJGI4XQIKCSEHK7DSXEAS4TSBV7IJANCNFSM4QFNWKQQ>
> .
>


-- 
Venita Hagerty, PhD
Research Associate
CIRA at NOAA, GSL/AVID
2B152, x6098

venitahagerty commented:
Okay, I made develop the default branch and merged the pull request into
develop. I deleted the master branch and a few older branches. There are
still a few extra feature branches. Should I click on the New pull request
button for those branches? Or delete them?

Thank you,
Venita

On Thu, Aug 20, 2020 at 2:21 PM Venita Hagerty - NOAA Affiliate <
venita.hagerty@noaa.gov> wrote:

> I've been trying to model what I do after the METviewer repository. It
> looks like that's transitioning to what you're describing. Is this
> described in the developer notes somewhere so I can refer back to it?
>
> I think I understand, and it sounds clearer without a single master
> branch. When I do a release, I make sure the develop branch is completely
> ready, and then I make a copy of it named as master with the release
> number. Over time, as I have multiple releases, I'll wind up with multiple
> master+release# branches. And I could remove the current master branch, and
> make develop the default.
>
> Thank you,
> Venita
>
> On Thu, Aug 20, 2020 at 2:09 PM johnhg <notifications@github.com> wrote:
>
>> Venita,
>>
>> I did see that my sed commands had changed some names like that. And I
>> wasn't totally sure about that.
>>
>> And I totally understand not using PR's when you're working alone. No need
>> really. I just wanted to submit my changes in this way so that you'd
>> review
>> them.
>>
>> I'm glad to see you have a "develop" branch in there. Let me redo this PR
>> with a branch from develop instead.
>>
>> And to avoid confusion, can we please just remove the master branch? We
>> don't have a master in the other METplus repos (or at least we shouldn't).
>> Instead, we have develop for new work going into the next release, and
>> master_vX.Y for each stable release. So when you do a v1.0 release, you'd
>> copy develop over into master_v1.0 and then you'd commit any bufixes for
>> v1.0 into both master_v1.0 (for a potential v1.0.1 bugfix release) and the
>> develop branch (for the next version).
>>
>> Does that all make sense?
>>
>> Thanks,
>> John
>>
>> On Thu, Aug 20, 2020 at 1:31 PM Venita Hagerty <notifications@github.com>
>> wrote:
>>
>> > I also hadn't updated to Master for a while, and I'm wondering if you
>> used
>> > master as your base?
>> >
>> > Venita
>> >
>> > On Thu, Aug 20, 2020 at 1:22 PM Venita Hagerty - NOAA Affiliate <
>> > venita.hagerty@noaa.gov> wrote:
>> >
>> > > Hi John,
>> > >
>> > > Since I was the only person on the repository for a while, and since I
>> > > haven't worked with Pull requests before, I haven't been using them. I
>> > may
>> > > need help with that.
>> > >
>> > > I have been making feature branches with the issue number, but I
>> didn't
>> > > realize you'd made one for renaming the repository.
>> > >
>> > > Also, I didn't think we were renaming METdbLoad.py. I thought the two
>> > > teams discussed that, and Tara was going to tell us if we needed to,
>> but
>> > > she wasn't expecting that we would.
>> > >
>> > > Thank you,
>> > > Venita
>> > >
>> > > On Thu, Aug 20, 2020 at 12:06 PM johnhg <notifications@github.com>
>> > wrote:
>> > >
>> > >> Venita,
>> > >>
>> > >> Great. If you're happy with the changes, can you please approve the
>> Pull
>> > >> Request? And then you could click the "Squash and Merge" button which
>> > will
>> > >> merge these changes into your master branch.
>> > >>
>> > >> This is the process we've been following for development in the other
>> > >> METplus repos.
>> > >> (1) Write up an issue to describe the new work.
>> > >> (2) Create a feature branch for that work that includes the issue
>> number
>> > >> (e.g. feature_33_github).
>> > >> (3) Do the work in the feature branch.
>> > >> (4) Submit a pull request to merge those changes back into the source
>> > >> branch.
>> > >> (5) Once the PR is approved, merge the changes and delete the feature
>> > >> branch.
>> > >>
>> > >> So we're at steps (4) and (5) here. If you'd prefer to meet through
>> > Google
>> > >> and share screens for this, please just let me know.
>> > >>
>> > >> Thanks,
>> > >> John
>> > >>
>> > >> On Thu, Aug 20, 2020 at 11:29 AM Venita Hagerty <
>> > notifications@github.com
>> > >> >
>> > >> wrote:
>> > >>
>> > >> > I have reviewed the files in dtcenter/METdatadb. I updated some
>> README
>> > >> > files and some comments in code. It looks good to me.
>> > >> >
>> > >> > On Wed, Aug 19, 2020 at 4:45 PM johnhg <notifications@github.com>
>> > >> wrote:
>> > >> >
>> > >> > > @JohnHalleyGotway <https://github.com/JohnHalleyGotway>
>> requested
>> > >> your
>> > >> > > review on: #34 <https://github.com/dtcenter/METdatadb/pull/34>
>> > >> Feature
>> > >> > 33
>> > >> > > GitHub.
>> > >> > >
>> > >> > > —
>> > >> > > You are receiving this because your review was requested.
>> > >> > > Reply to this email directly, view it on GitHub
>> > >> > > <https://github.com/dtcenter/METdatadb/pull/34#event-3672860800
>> >,
>> > or
>> > >> > > unsubscribe
>> > >> > > <
>> > >> >
>> > >>
>> >
>> https://github.com/notifications/unsubscribe-auth/AJGI4XVWZZAUGB5R6MQ4F6TSBRIZFANCNFSM4QFNWKQQ
>> > >> > >
>> > >> > > .
>> > >> > >
>> > >> >
>> > >> >
>> > >> > --
>> > >> > Venita Hagerty, PhD
>> > >> > Research Associate
>> > >> > CIRA at NOAA, GSL/AVID
>> > >> > 2B152, x6098
>> > >> >
>> > >> > —
>> > >> > You are receiving this because you were mentioned.
>> > >> > Reply to this email directly, view it on GitHub
>> > >> > <
>> https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677798966
>> > >,
>> > >> > or unsubscribe
>> > >> > <
>> > >>
>> >
>> https://github.com/notifications/unsubscribe-auth/AFA4HKD4R76WPSIR47VUZLTSBVMQHANCNFSM4QFNWKQQ
>> > >> >
>> > >> > .
>> > >> >
>> > >>
>> > >> —
>> > >> You are receiving this because your review was requested.
>> > >> Reply to this email directly, view it on GitHub
>> > >> <
>> https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677816749>,
>> > >> or unsubscribe
>> > >> <
>> >
>> https://github.com/notifications/unsubscribe-auth/AJGI4XW62GJQLK7XVPF6UALSBVQ3DANCNFSM4QFNWKQQ
>> > >
>> > >> .
>> > >>
>> > >
>> > >
>> > > --
>> > > Venita Hagerty, PhD
>> > > Research Associate
>> > > CIRA at NOAA, GSL/AVID
>> > > 2B152, x6098
>> > >
>> >
>> >
>> > --
>> > Venita Hagerty, PhD
>> > Research Associate
>> > CIRA at NOAA, GSL/AVID
>> > 2B152, x6098
>> >
>> > —
>> > You are receiving this because you were mentioned.
>> > Reply to this email directly, view it on GitHub
>> > <https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677857757>,
>> > or unsubscribe
>> > <
>> https://github.com/notifications/unsubscribe-auth/AFA4HKCJQUS7UZ7KGNWFPUDSBV2YHANCNFSM4QFNWKQQ
>> >
>> > .
>> >
>>
>> —
>> You are receiving this because your review was requested.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677877914>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AJGI4XQIKCSEHK7DSXEAS4TSBV7IJANCNFSM4QFNWKQQ>
>> .
>>
>
>
> --
> Venita Hagerty, PhD
> Research Associate
> CIRA at NOAA, GSL/AVID
> 2B152, x6098
>


-- 
Venita Hagerty, PhD
Research Associate
CIRA at NOAA, GSL/AVID
2B152, x6098

JohnHalleyGotway commented:
Venita,

The intended workflow is listed in the "checklist" at the bottom of each
GitHub issue template (e.g. feature branch naming conventions, etc). So you
could look there for a brief overview.

The git workflow is described in more detail here:
https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html

And I actually have a GitHub issue to describe the pull request approval
process in that git workflow (https://github.com/dtcenter/METplus/issues/429
).
But I haven't gotten to that yet.

As for what to do with your existing feature branches, if those changes
have already been merged into develop, I'd recommend deleting them. If not,
then it's up to you.

FYI, I've also been meeting with Keith Searight and other METplus folks
about how to use TravisCI across all of our repos. The goal is so that when
someone submits a pull request to change develop or a master_vX.Y branch,
that'd automatically trigger Travis to run a bunch of tests to make sure
those changes don't break anything. So hopefully, we'll eventually be able
to integrate that into all of our repos. But I do realize that figuring out
exactly what tests should be run is a big question!

Thanks,
John

On Thu, Aug 20, 2020 at 2:41 PM Venita Hagerty <notifications@github.com>
wrote:

> Okay, I made develop the default branch and merged the pull request into
> develop. I deleted the master branch and a few older branches. There are
> still a few extra feature branches. Should I click on the New pull request
> button for those branches? Or delete them?
>
> Thank you,
> Venita
>
> On Thu, Aug 20, 2020 at 2:21 PM Venita Hagerty - NOAA Affiliate <
> venita.hagerty@noaa.gov> wrote:
>
> > I've been trying to model what I do after the METviewer repository. It
> > looks like that's transitioning to what you're describing. Is this
> > described in the developer notes somewhere so I can refer back to it?
> >
> > I think I understand, and it sounds clearer without a single master
> > branch. When I do a release, I make sure the develop branch is completely
> > ready, and then I make a copy of it named as master with the release
> > number. Over time, as I have multiple releases, I'll wind up with
> multiple
> > master+release# branches. And I could remove the current master branch,
> and
> > make develop the default.
> >
> > Thank you,
> > Venita
> >
> > On Thu, Aug 20, 2020 at 2:09 PM johnhg <notifications@github.com> wrote:
> >
> >> Venita,
> >>
> >> I did see that my sed commands had changed some names like that. And I
> >> wasn't totally sure about that.
> >>
> >> And I totally understand not using PR's when you're working alone. No
> need
> >> really. I just wanted to submit my changes in this way so that you'd
> >> review
> >> them.
> >>
> >> I'm glad to see you have a "develop" branch in there. Let me redo this
> PR
> >> with a branch from develop instead.
> >>
> >> And to avoid confusion, can we please just remove the master branch? We
> >> don't have a master in the other METplus repos (or at least we
> shouldn't).
> >> Instead, we have develop for new work going into the next release, and
> >> master_vX.Y for each stable release. So when you do a v1.0 release,
> you'd
> >> copy develop over into master_v1.0 and then you'd commit any bufixes for
> >> v1.0 into both master_v1.0 (for a potential v1.0.1 bugfix release) and
> the
> >> develop branch (for the next version).
> >>
> >> Does that all make sense?
> >>
> >> Thanks,
> >> John
> >>
> >> On Thu, Aug 20, 2020 at 1:31 PM Venita Hagerty <
> notifications@github.com>
> >> wrote:
> >>
> >> > I also hadn't updated to Master for a while, and I'm wondering if you
> >> used
> >> > master as your base?
> >> >
> >> > Venita
> >> >
> >> > On Thu, Aug 20, 2020 at 1:22 PM Venita Hagerty - NOAA Affiliate <
> >> > venita.hagerty@noaa.gov> wrote:
> >> >
> >> > > Hi John,
> >> > >
> >> > > Since I was the only person on the repository for a while, and
> since I
> >> > > haven't worked with Pull requests before, I haven't been using
> them. I
> >> > may
> >> > > need help with that.
> >> > >
> >> > > I have been making feature branches with the issue number, but I
> >> didn't
> >> > > realize you'd made one for renaming the repository.
> >> > >
> >> > > Also, I didn't think we were renaming METdbLoad.py. I thought the
> two
> >> > > teams discussed that, and Tara was going to tell us if we needed to,
> >> but
> >> > > she wasn't expecting that we would.
> >> > >
> >> > > Thank you,
> >> > > Venita
> >> > >
> >> > > On Thu, Aug 20, 2020 at 12:06 PM johnhg <notifications@github.com>
> >> > wrote:
> >> > >
> >> > >> Venita,
> >> > >>
> >> > >> Great. If you're happy with the changes, can you please approve the
> >> Pull
> >> > >> Request? And then you could click the "Squash and Merge" button
> which
> >> > will
> >> > >> merge these changes into your master branch.
> >> > >>
> >> > >> This is the process we've been following for development in the
> other
> >> > >> METplus repos.
> >> > >> (1) Write up an issue to describe the new work.
> >> > >> (2) Create a feature branch for that work that includes the issue
> >> number
> >> > >> (e.g. feature_33_github).
> >> > >> (3) Do the work in the feature branch.
> >> > >> (4) Submit a pull request to merge those changes back into the
> source
> >> > >> branch.
> >> > >> (5) Once the PR is approved, merge the changes and delete the
> feature
> >> > >> branch.
> >> > >>
> >> > >> So we're at steps (4) and (5) here. If you'd prefer to meet through
> >> > Google
> >> > >> and share screens for this, please just let me know.
> >> > >>
> >> > >> Thanks,
> >> > >> John
> >> > >>
> >> > >> On Thu, Aug 20, 2020 at 11:29 AM Venita Hagerty <
> >> > notifications@github.com
> >> > >> >
> >> > >> wrote:
> >> > >>
> >> > >> > I have reviewed the files in dtcenter/METdatadb. I updated some
> >> README
> >> > >> > files and some comments in code. It looks good to me.
> >> > >> >
> >> > >> > On Wed, Aug 19, 2020 at 4:45 PM johnhg <notifications@github.com
> >
> >> > >> wrote:
> >> > >> >
> >> > >> > > @JohnHalleyGotway <https://github.com/JohnHalleyGotway>
> >> requested
> >> > >> your
> >> > >> > > review on: #34 <https://github.com/dtcenter/METdatadb/pull/34>
> >> > >> Feature
> >> > >> > 33
> >> > >> > > GitHub.
> >> > >> > >
> >> > >> > > —
> >> > >> > > You are receiving this because your review was requested.
> >> > >> > > Reply to this email directly, view it on GitHub
> >> > >> > > <
> https://github.com/dtcenter/METdatadb/pull/34#event-3672860800
> >> >,
> >> > or
> >> > >> > > unsubscribe
> >> > >> > > <
> >> > >> >
> >> > >>
> >> >
> >>
> https://github.com/notifications/unsubscribe-auth/AJGI4XVWZZAUGB5R6MQ4F6TSBRIZFANCNFSM4QFNWKQQ
> >> > >> > >
> >> > >> > > .
> >> > >> > >
> >> > >> >
> >> > >> >
> >> > >> > --
> >> > >> > Venita Hagerty, PhD
> >> > >> > Research Associate
> >> > >> > CIRA at NOAA, GSL/AVID
> >> > >> > 2B152, x6098
> >> > >> >
> >> > >> > —
> >> > >> > You are receiving this because you were mentioned.
> >> > >> > Reply to this email directly, view it on GitHub
> >> > >> > <
> >> https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677798966
> >> > >,
> >> > >> > or unsubscribe
> >> > >> > <
> >> > >>
> >> >
> >>
> https://github.com/notifications/unsubscribe-auth/AFA4HKD4R76WPSIR47VUZLTSBVMQHANCNFSM4QFNWKQQ
> >> > >> >
> >> > >> > .
> >> > >> >
> >> > >>
> >> > >> —
> >> > >> You are receiving this because your review was requested.
> >> > >> Reply to this email directly, view it on GitHub
> >> > >> <
> >> https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677816749>,
> >> > >> or unsubscribe
> >> > >> <
> >> >
> >>
> https://github.com/notifications/unsubscribe-auth/AJGI4XW62GJQLK7XVPF6UALSBVQ3DANCNFSM4QFNWKQQ
> >> > >
> >> > >> .
> >> > >>
> >> > >
> >> > >
> >> > > --
> >> > > Venita Hagerty, PhD
> >> > > Research Associate
> >> > > CIRA at NOAA, GSL/AVID
> >> > > 2B152, x6098
> >> > >
> >> >
> >> >
> >> > --
> >> > Venita Hagerty, PhD
> >> > Research Associate
> >> > CIRA at NOAA, GSL/AVID
> >> > 2B152, x6098
> >> >
> >> > —
> >> > You are receiving this because you were mentioned.
> >> > Reply to this email directly, view it on GitHub
> >> > <https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677857757
> >,
> >> > or unsubscribe
> >> > <
> >>
> https://github.com/notifications/unsubscribe-auth/AFA4HKCJQUS7UZ7KGNWFPUDSBV2YHANCNFSM4QFNWKQQ
> >> >
> >> > .
> >> >
> >>
> >> —
> >> You are receiving this because your review was requested.
> >> Reply to this email directly, view it on GitHub
> >> <https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677877914>,
> >> or unsubscribe
> >> <
> https://github.com/notifications/unsubscribe-auth/AJGI4XQIKCSEHK7DSXEAS4TSBV7IJANCNFSM4QFNWKQQ
> >
> >> .
> >>
> >
> >
> > --
> > Venita Hagerty, PhD
> > Research Associate
> > CIRA at NOAA, GSL/AVID
> > 2B152, x6098
> >
>
>
> --
> Venita Hagerty, PhD
> Research Associate
> CIRA at NOAA, GSL/AVID
> 2B152, x6098
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METdatadb/pull/34#issuecomment-677892981>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFA4HKHVGPBBMOVFO43VSV3SBWDBBANCNFSM4QFNWKQQ>
> .
>
